gu-net/src/rpc/peer.rs

```diff
@@ -62,16 +68,54 @@ impl Message for UpdatePeer {
 
 pub struct PeerManager {
     peers: HashMap<NodeId, PeerInfo>,
+    path: PathBuf,
+    saved_tags: HashMap<NodeId, Tags>,
 }
 
 impl Actor for PeerManager {
     type Context = Context<Self>;
+
+    fn started(&mut self, _ctx: &mut <Self as Actor>::Context) {
+        self.path = ConfigModule::new().work_dir().join("tags");
+        // FIXME this one is awful
+        let tags_serialized = fs::read_to_string(&self.path).unwrap_or_else(|e| match e.kind() {
+            io::ErrorKind::NotFound => "{}".into(),
+            _ => panic!(
```

Comment:
What's better? 1) panic!(...) or 2) error!(..) and continue.

Reply:
`panic!` - if there are some other problems with permissions/filesystem, they should be fixed by the administrator, or users will start getting problems.